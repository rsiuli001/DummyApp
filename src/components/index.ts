export { default as ScreenList } from './ScreenList/ScreenList';
export { default as Buttons } from './Buttons/Buttons';
export { default as Banner } from './Banner/Banner';
