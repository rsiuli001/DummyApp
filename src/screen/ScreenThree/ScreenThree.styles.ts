import { StyleSheet } from 'react-native';

const ScreenThreeStyles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    marginTop: 20,
  },
  textContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center',
    marginRight: 20,
    marginBottom: 10,
  },
  text: {
    fontWeight: '400',
    fontSize: 16,
  },
});

export default ScreenThreeStyles;
