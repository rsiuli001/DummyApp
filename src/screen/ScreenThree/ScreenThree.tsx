import React, { FC } from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';
import { ScreenList } from '../../components';
import ScreenThreeStyles from './ScreenThree.styles';
import * as ScreenName from '../../constants/screenName';
import * as Strings from '../../constants/strings';
import * as Colors from '../../constants/colors';

interface ScreenThreeProps {
  userName: string;
  navigation: any;
}

export const ScreenThree: FC<ScreenThreeProps> = ({
  userName,
  navigation,
}): JSX.Element => {
  const onPress = (screenName: string) => {
    navigation.navigate(screenName);
  };

  return (
    <View style={ScreenThreeStyles.container}>
      <View style={ScreenThreeStyles.textContainer}>
        <Text style={ScreenThreeStyles.text}>
          {Strings.WELCOME} {userName}
        </Text>
      </View>

      <ScreenList
        text={`${Strings.NAVIGATE_TO} ${Strings.SCREEN_ONE}`}
        color={Colors.BLACK}
        onPress={onPress}
        screenName={ScreenName.SCREEN_ONE}
      />
      <ScreenList
        text={`${Strings.NAVIGATE_TO} ${Strings.SCREEN_TWO}`}
        color={Colors.BLACK}
        onPress={onPress}
        screenName={ScreenName.SCREEN_TWO}
      />
      <ScreenList
        text={`${Strings.NAVIGATE_TO} ${Strings.SCREEN_THREE}`}
        color={Colors.BLACK}
        onPress={onPress}
        screenName={ScreenName.SCREEN_THREE}
      />
    </View>
  );
};

const mapStateToProps = (state: ScreenThreeProps) => ({
  userName: state.userName,
});

export default connect(mapStateToProps)(ScreenThree);
