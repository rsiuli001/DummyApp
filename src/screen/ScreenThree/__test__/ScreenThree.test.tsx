import React from 'react';
import { shallow } from 'enzyme';
import { ScreenThree } from '../ScreenThree';
import { View } from 'react-native';
import { ScreenList } from '../../../components';

describe('Screen three Tests', () => {
  describe('rendering screen one', () => {
    const wrapper = shallow(
      <ScreenThree userName="userName" navigation={{}} />
    );

    it('should render view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(View).length).toEqual(2);
    });

    it('should render ScreenList component', () => {
      expect(wrapper.find(ScreenList).length).toEqual(3);
    });
  });
});
