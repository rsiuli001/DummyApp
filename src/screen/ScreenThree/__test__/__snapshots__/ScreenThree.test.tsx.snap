// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen three Tests rendering screen one should render view 1`] = `ShallowWrapper {}`;
